### PR TITLE
Develop

### PR DESCRIPTION
**Pull request by phlyash**

### CODE REVIEW

File: currency.py [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+
+
+def leap_year_check(year):
+    if year % 4 == 0 and (year % 400 == 0 or year % 100 != 0):
+        return True
+    else:
+        return False
+
+
+def days_in_month(month, year):
```
Comment (jacob1237) [inline currency.py]: You've hit the nail on the date parsing algorithms: they can be really complicated. I see that you've tried to detect the number of days in February depending on the year, etc.

However, Python already contains that stuff in the standard library. Here is how to do that:

```python
from datetime import datetime

print(datetime.strptime("10/01/2020", "%d/%m/%Y"))
```

So this function will parse an arbitrary date string for you depending on the date format you've passed as the second argument. Here is the complete list of available date format placeholders: https://docs.python.org/3/library/datetime.html#strftime-and-strptime-format-codes

In our current situation, you need only three of them: `%d` (days in format 01-31), `%m` (months in format 1-12), and `%Y` which represents the year with four numbers.

This function returns the `datetime` object, which contains information about the month, day, year, hours, seconds, etc. as separate variables. For example, here is how you can get the year from it:

```python
from datetime import datetime

dt = datetime.strptime("2020/01/01", "%Y/%m/%d")
print(dt.year)
```
Once you've got the `datetime` object, you can convert it back to a string representation using the specific date format again. For example:

```python
from datetime import datetime

dt = datetime.strptime("2020/01/01", "%Y/%m/%d")
print(dt.strftime("%B %d, %Y")) # will give you 'January 01, 2020'
```

To summarize, there are two functions:
1. `datetime.strptime()` which can parse your date
2. `datetime.strftime()` which can convert your datetime object to a string in accordance with specific date format

Try to use them instead of your own code

File: currency.py [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+
+if len(sys.argv) == 1 or sys.argv[1] == "-h" or sys.argv[1] == "-a":
+    print(
+        "usage: currency.py [valute code] [date: <format{yyyy-mm-dd}>]", "\n",
+        "usage example: eur 2020-01-01", "\n",
+        "this script has data from 1992-07-01", "\n",
+        "other arguments: [-h] [-a] [-ec] [-exit_codes]", sep=""
+          )
+    sys.exit("exit code: 3")
+elif sys.argv[1] == "-exit_codes" or sys.argv[1] == "-ec":
```
Comment (jacob1237) [inline currency.py]: It's not that common to use long names for short command-line options. Usually, when you use the `-` syntax instead of `--`, you have to use a single letter. Of course, it's possible to keep it as you made, but it's better to stay compliant with other command-line tools.

Here is what I propose: replace `-exit_codes` and `-ec` with `-e` and `--exit-codes`

File: currency.py [hunk 3]
```diff
@@ -0,0 +1,83 @@
...
+
+
+if len(sys.argv) == 1 or sys.argv[1] == "-h" or sys.argv[1] == "-a":
+    print(
+        "usage: currency.py [valute code] [date: <format{yyyy-mm-dd}>]", "\n",
+        "usage example: eur 2020-01-01", "\n",
+        "this script has data from 1992-07-01", "\n",
+        "other arguments: [-h] [-a] [-ec] [-exit_codes]", sep=""
+          )
+    sys.exit("exit code: 3")
```
Comment (jacob1237) [inline currency.py]: While you've got the concept of exit codes in command-line programs, the real exit codes are usually defined as integers. At least this is how most operation systems treat them.

You can still use a string in `sys.exit()`, but there is no reason to pass the number there because the string is for humans while integers are for the computer.

So if you decide to return a string, it's better to write the complete error message, like this:
```python
sys.exit("Currency not found")
```
but if you want integer codes, use them correctly:
```python
ERROR_CURRENCY = 2 # this constant must be defined at the beginning of the program

sys.exit(ERROR_CURRENCY)
```

File: currency.py [hunk 4]
```diff
@@ -0,0 +1,83 @@
...
+elif sys.argv[1] == "-exit_codes" or sys.argv[1] == "-ec":
+    print(
+        "exit code 0: success", "\n",
+        "exit code 1: currency-code format error", "\n",
+        "exit code 2: date error(not in data base or date in future)", "\n",
+        "exit code 3: references", sep=""
+    )
+    sys.exit("exit code: 3")
+currency = sys.argv[1]
```
Comment (jacob1237) [inline currency.py]: I would recommend dividing some code blocks with empty lines:

```python
    # ...
    sys.exit("exit code: 3")

currency = sys.argv[1]
date = sys.argv[2] # y-m-d

if date_check(date):
   # ...
```

While this is not in the standard, it makes it easier to read the code!

File: currency.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+date = sys.argv[2] # y-m-d
+if date_check(date):
+    lst = date.split("-")  # d/m/y
+    lst[0], lst[2] = lst[2], lst[0]  # swap day and year for cbr format
+    date = "/".join(lst)  # date for cbr
+    url = "http://www.cbr.ru/scripts/XML_daily.asp?date_req=" + date
+    html = str(urllib.request.urlopen(url).read())
+    if currency.upper() in html:
```
Comment (jacob1237) [inline currency.py]: You're calling the `upper()` function four times here. While this is not critical for our current example, it's always better to store the uppercase value in a separate variable and then reuse it, because some functions may be computationally expensive:
```python
currency = sys.argv[1].upper()

if currency in html:
    # ...
```

File: currency.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+        print(f"Курс рубля к {currency.upper()} на {date}: "
+              f"{html[html.find('Value', html.find(currency.upper())) + 6:html.find('Value',html.find(currency.upper())) + 13]}")
+        sys.exit("exit code: 0")
+    else:
+        print(f"error: currency({currency}) not found")
```
Comment (jacob1237) [inline currency.py]: You don't need the `print` function here. You can just use `sys.exit(f"error: currency({currency}) not found")`

**Review by jacob1237 — CHANGES_REQUESTED** [review]
Requesting changes
